org.osgi.util.converter/src/org/osgi/util/converter/DTOUtil.java

```diff
@@ -18,45 +18,53 @@
 
 package org.osgi.util.converter;
 
+import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
 import java.lang.reflect.Method;
 import java.lang.reflect.Modifier;
+import java.util.Arrays;
 
 /**
  * @author $Id$
  */
 class DTOUtil {
+	private static final Method[] OBJECT_CLASS_METHODS = Object.class
+			.getMethods();
+
 	private DTOUtil() {
 		// Do not instantiate. This is a utility class.
 	}
 
 	static boolean isDTOType(Class< ? > cls, boolean ignorePublicNoArgsCtor) {
 		if (!ignorePublicNoArgsCtor) {
-			try {
-				cls.getConstructor();
-			} catch (NoSuchMethodException | SecurityException e) {
-				// No public zero-arg constructor, not a DTO
+			boolean hasPublicNoargsCtor = false;
+			for (Constructor< ? > ctor : cls.getConstructors()) {
+				if (ctor.getParameterCount() == 0) {
+					hasPublicNoargsCtor = true;
+					break;
+				}
+			}
+			if (!hasPublicNoargsCtor) {
```

Comment:
```suggestion
			if (Arrays.stream(cls.getConstructors())
					.noneMatch(ctor -> ctor.getParameterCount() == 0)) {
				// No public zero-arg constructor, not a DTO
```
More concise alternative.